controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -403,15 +403,8 @@ public Void initializeStreamWriters(final String commitStreamName, final EventSt
             return CompletableFuture.completedFuture(createStatus(Status.DISCONNECTED));
         }
 
-        if (timeoutService.containsTxn(scope, stream, txnId)) {
-            // If timeout service knows about this transaction, attempt to increase its lease.
-            log.debug("Txn={}, extending lease in timeout service", txnId);
-            return CompletableFuture.completedFuture(timeoutService.pingTxn(scope, stream, txnId, lease));
-        } else {
-            // Otherwise, fence other potential processes managing timeout for this txn, and update its lease.
-            log.debug("Txn={}, updating txn node in store and extending lease", txnId);
-            return fenceTxnUpdateLease(scope, stream, txnId, lease, ctx);
-        }
+        log.debug("Txn={}, updating txn node in store and extending lease", txnId);
+        return fenceTxnUpdateLease(scope, stream, txnId, lease, ctx);
```

Comment:
To confirm what this is doing: this call will update the txn znode version so that a different controller instance can't complete this transaction, unless it also does another znode version update. If my interpretation is correct, then this is the correct behavior because we can't completely prevent other controller instances from thinking that they own the txn. The best we can do is to fence the txn via zk so that only the latest owner (determined by the znode version) can make changes to the txn.

Reply:
Yes

Reply:
Ok, I thought some more about this, and the approach seems to be overkill as I understand it. The first time a controller instance sees the ping for a txn, it should definitely fence it, but not on every ping. This PR is inducing one write to zk upon every ping, which is unnecessary. Instead, we need to make sure that a controller instance holds the lease when it decides to commit /abort a txn. The way to achieve this is to conditionally update the status of a txn from `OPEN` to `COMMITTING\ABORTING`, and the update must be conditional on the version obtained when the controller instance fenced the txn upon the first ping.

Also, the javadoc for this method seems to be stale when contrasted with the changes here.

Reply:
that is exactly how the code was prior to this fix. 
the problem with that is that controllers can receive pings interspersed with each other. And can erroneously assume that they dont need to fence.. 

For example
C1  receives ping, fences [txn version 1], starts timer for txn
C2 receives ping, fences [txn version 2] , starts timer for txn
C1 receives ping, no fencing[txn version 1], resets timer, tracks txn in memory
C1 receives more pings and [txn version 1] continues to track in memory.
C2 finds timer timedout, [txn version 2] attempts abort and succeeds. 


So one solution that i can think of, on presumption that txn ping will be sent by one client only:
Add a version to the ping. 
C1 receives ping[1], fences [txn version 1], starts timer
C2 receives ping[2], fences [txn version 2], starts timer
C1 receives ping[3], compares with in-memory state and knows it had a stale value in memory. fences[txn version 3]. resets timer and tracks
C2 receives ping[4], compares, fences [txn version 4], resets
C1 receives ping[5], compares, fences [txn version 5], resets
C1 receives ping[6], [txn version 5] resets in memory timer
C2 times out, [txn version 4] attempts abort, fails with write conflict

Reply:
Let's go with this approach for now and revisit it in issue #1838 .